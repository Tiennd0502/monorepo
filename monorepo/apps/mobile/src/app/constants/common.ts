export const CONGRATS_IMAGE = 'https://i.ibb.co/nkf6hFr/congrats.png';
