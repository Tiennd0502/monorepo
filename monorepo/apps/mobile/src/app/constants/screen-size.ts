import { Dimensions } from 'react-native';

export const BASE_SCREEN_WIDTH = 412;
export const BASE_SCREEN_HEIGHT = 842;

export const SCREEN_WIDTH: number = Dimensions.get('window').width;
export const SCREEN_HEIGHT: number = Dimensions.get('window').height;

export const CONTENT_WIDTH: number = SCREEN_WIDTH - 40;
