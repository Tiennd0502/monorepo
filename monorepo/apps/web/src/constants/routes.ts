export const ROUTES = {
  HOME: '/',
  VERIFY_OTP: '/verify-otp',
  SIGN_UP: '/sign-up',
  LOGIN: '/login',
};
