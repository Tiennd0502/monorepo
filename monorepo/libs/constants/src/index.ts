export * from './apis';
export * from './message';
export * from './commons';
export * from './product';
export * from './category';
export * from './validation';
