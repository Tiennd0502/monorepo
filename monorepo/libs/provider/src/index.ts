export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
