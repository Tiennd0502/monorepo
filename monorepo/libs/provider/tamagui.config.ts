import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
