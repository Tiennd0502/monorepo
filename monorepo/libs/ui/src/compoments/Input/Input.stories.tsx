import { Meta, StoryObj } from '@storybook/react';

import Input from '.';

const meta: Meta<typeof Input> = {
  title: 'Components/Input',
  component: Input,
  args: {
    paddingHorizontal: 20,
  },
};

export default meta;

type Story = StoryObj<typeof Input>;

export const Default: Story = {
  args: {
    label: 'Username:',
    placeholder: 'Enter username',
  },
};

export const OutLined: Story = {
  args: {
    variant: 'outlined',
    label: 'Username:',
    placeholder: 'Enter username',
  },
};

export const Disabled: Story = {
  args: {
    disabled: true,
    label: 'Username:',
    placeholder: 'Enter username',
  },
};

export const Flushed: Story = {
  args: {
    variant: 'flushed',
    label: 'Username:',
    placeholder: 'Enter username',
  },
};
