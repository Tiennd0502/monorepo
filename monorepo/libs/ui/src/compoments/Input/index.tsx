import {
  styled,
  InputProps as TInputProps,
  Input as TInput,
  XStack,
  ButtonProps,
} from 'tamagui';
import { memo, useMemo } from 'react';

import Text from '../Text';
import IconButton from '../IconButton';

const StyledInput = styled(TInput, {
  flex: 1,
  borderWidth: '$px',
  borderRadius: '$small',
  color: '$textDefault',
  paddingHorizontal: '$4',
  paddingTop: '$6.5',
  paddingBottom: '$3.5',
  height: '$12.5',
  placeholderTextColor: '$textDefault',
  allowFontScaling: false,

  variants: {
    solid: {
      true: {
        backgroundColor: '$primary',
        color: '$textSecondary',
        borderColor: '$primary',
      },
    },

    outlined: {
      true: {
        backgroundColor: '$secondary',
        borderColor: '$borderSecondary',
        color: 'textPrimary',
      },
    },

    flushed: {
      true: {
        backgroundColor: 'transparent',
        color: 'textPrimary',
        borderColor: 'transparent',
        focusStyle: {
          borderColor: 'transparent',
        },
      },
    },

    disabled: {
      true: {
        backgroundColor: '$backgroundInputDisabled',
        borderColor: '$backgroundInputDisabled',
        color: 'textPlaceholder',
      },
    },

    size: {
      sm: {
        height: '$11',
        paddingVertical: '$3',
      },

      md: {
        fontSize: '$4',
        height: '$16.5',
      },
    },
  } as const,

  defaultVariants: {
    solid: true,
    size: 'md',
  },
});

const StyledLabel = styled(Text, {
  position: 'absolute',
  zIndex: '$2',
  top: '$2',
  left: '$4',
  color: '$textLabel',

  variants: {
    size: {
      md: {
        fontSize: '$3',
      },
    },
  } as const,

  defaultVariants: {
    size: 'md',
  },
});

const StyledErrorMessage = styled(Text, {
  position: 'absolute',
  zIndex: '$2',
  bottom: 0,
  left: 0,
  marginBottom: -20,

  variants: {
    size: {
      md: {
        fontSize: '$3',
      },
    },
  } as const,
});

interface InputProps extends Omit<TInputProps, 'variant' | 'size'> {
  label?: string;
  variant?: 'outlined' | 'solid' | 'flushed' | 'disabled';
  size?: 'sm' | 'md';
  rightElement?: ButtonProps;
  errorMessage?: string;
}

const Input = StyledInput.styleable<InputProps>(
  (
    {
      size = 'md',
      variant = 'solid',
      label = '',
      errorMessage = '',
      rightElement,
      ...rest
    },
    ref
  ) => {
    const variantObject = useMemo(
      () => (variant ? { [variant]: true } : {}),
      [variant]
    );

    return (
      <XStack position="relative" zIndex={1} alignItems="center">
        {label && <StyledLabel>{label}</StyledLabel>}
        <StyledInput ref={ref} size={size} {...variantObject} {...rest} />
        {rightElement && (
          <IconButton
            aria-label="Icon"
            backgroundColor="transparent"
            padding={0}
            right="$4"
            hoverStyle={{
              backgroundColor: 'transparent',
            }}
            pressStyle={{
              backgroundColor: 'transparent',
            }}
            position="absolute"
            {...rightElement}
          />
        )}
        {errorMessage && (
          <StyledErrorMessage error>{errorMessage}</StyledErrorMessage>
        )}
      </XStack>
    );
  }
);

export default memo(Input);
