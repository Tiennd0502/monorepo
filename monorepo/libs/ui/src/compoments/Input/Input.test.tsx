import {fireEvent, render, waitFor} from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import { ERROR_MESSAGES } from '@monorepo/constants';
import Input from '.';

const baseProps = {
  label: 'Name',
  placeholder: 'Enter value',
};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <Input {...props} />
    </TamaguiProvider>
  );
};

describe('Input', () => {

  it('should render Input component with default props', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });

  it('should render Input component with variant outlined', () => {
    const container = setup({variant: "outlined"});

    expect(container).toMatchSnapshot();
  });

  it('should render Input component with variant flushed', () => {
    const container = setup({variant: "flushed"});

    expect(container).toMatchSnapshot();
  });

  it('should render Input component with variant disabled', () => {
    const container = setup({variant: 'disabled'});

    expect(container).toMatchSnapshot();
  });

  it('should render Input component with error message', () => {
    const container =  setup({ errorMessage: ERROR_MESSAGES.FIELD_REQUIRED('Name')});

    expect(container).toMatchSnapshot();
  });

  it('should render Input component when entering input', () => {
    const onChangeText = jest.fn()
    const {getByTestId} = setup({onChangeText});
    
    waitFor(() => {
      const result = '123';
      const input = getByTestId('input');
      fireEvent.changeText(input, result);
  
      expect(onChangeText).toHaveBeenCalledWith(result);
    })
  });
});
