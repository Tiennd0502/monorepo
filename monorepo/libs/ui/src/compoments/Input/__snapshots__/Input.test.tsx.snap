// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render Input component with default props 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "position": "relative",
      "zIndex": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#808080",
        "fontFamily": "NunitoSans_7pt-Medium",
        "fontSize": 12,
        "left": 16,
        "position": "absolute",
        "top": 8,
        "zIndex": 200,
      }
    }
    suppressHighlighting={true}
  >
    Name
  </Text>
  <TextInput
    allowFontScaling={false}
    focusVisibleStyle={{}}
    focusable={true}
    onChangeText={[Function]}
    placeholder="Enter value"
    placeholderTextColor="#999"
    style={
      {
        "backgroundColor": "#fff",
        "borderBottomColor": "transparent",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "borderBottomWidth": 1,
        "borderLeftColor": "transparent",
        "borderLeftWidth": 1,
        "borderRightColor": "transparent",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "transparent",
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "borderTopWidth": 1,
        "color": "#242424",
        "flex": 1,
        "fontFamily": "NunitoSans_7pt-Medium",
        "fontSize": 16,
        "height": 66,
        "letterSpacing": -0.5,
        "minWidth": 0,
        "paddingBottom": 14,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 26,
      }
    }
    testID="input"
  />
</View>
`;

exports[`Input should render Input component with error message 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "position": "relative",
      "zIndex": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#808080",
        "fontFamily": "NunitoSans_7pt-Medium",
        "fontSize": 12,
        "left": 16,
        "position": "absolute",
        "top": 8,
        "zIndex": 200,
      }
    }
    suppressHighlighting={true}
  >
    Name
  </Text>
  <TextInput
    allowFontScaling={false}
    focusVisibleStyle={{}}
    focusable={true}
    onChangeText={[Function]}
    placeholder="Enter value"
    placeholderTextColor="#999"
    style={
      {
        "backgroundColor": "#fff",
        "borderBottomColor": "transparent",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "borderBottomWidth": 1,
        "borderLeftColor": "transparent",
        "borderLeftWidth": 1,
        "borderRightColor": "transparent",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "transparent",
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "borderTopWidth": 1,
        "color": "#242424",
        "flex": 1,
        "fontFamily": "NunitoSans_7pt-Medium",
        "fontSize": 16,
        "height": 66,
        "letterSpacing": -0.5,
        "minWidth": 0,
        "paddingBottom": 14,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 26,
      }
    }
    testID="input"
  />
  <Text
    allowFontScaling={false}
    style={
      {
        "bottom": 0,
        "color": "#eb5757",
        "fontFamily": "NunitoSans_7pt-Medium",
        "fontSize": 14,
        "left": 16,
        "lineHeight": 20,
        "marginBottom": -20,
        "position": "absolute",
        "zIndex": 200,
      }
    }
    suppressHighlighting={true}
  >
    Name is required
  </Text>
</View>
`;

exports[`Input should render Input component with variant disabled 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "position": "relative",
      "zIndex": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#808080",
        "fontFamily": "NunitoSans_7pt-Medium",
        "fontSize": 12,
        "left": 16,
        "position": "absolute",
        "top": 8,
        "zIndex": 200,
      }
    }
    suppressHighlighting={true}
  >
    Name
  </Text>
  <TextInput
    allowFontScaling={false}
    focusVisibleStyle={{}}
    focusable={true}
    onChangeText={[Function]}
    placeholder="Enter value"
    placeholderTextColor="#999"
    readOnly={true}
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderBottomColor": "#f5f5f5",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "borderBottomWidth": 1,
        "borderLeftColor": "#f5f5f5",
        "borderLeftWidth": 1,
        "borderRightColor": "#f5f5f5",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "#f5f5f5",
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "borderTopWidth": 1,
        "color": "textPlaceholder",
        "flex": 1,
        "fontFamily": "NunitoSans_7pt-Medium",
        "fontSize": 16,
        "height": 66,
        "letterSpacing": -0.5,
        "minWidth": 0,
        "paddingBottom": 14,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 26,
      }
    }
    testID="input"
  />
</View>
`;

exports[`Input should render Input component with variant flushed 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "position": "relative",
      "zIndex": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#808080",
        "fontFamily": "NunitoSans_7pt-Medium",
        "fontSize": 12,
        "left": 16,
        "position": "absolute",
        "top": 8,
        "zIndex": 200,
      }
    }
    suppressHighlighting={true}
  >
    Name
  </Text>
  <TextInput
    allowFontScaling={false}
    focusVisibleStyle={{}}
    focusable={true}
    onChangeText={[Function]}
    placeholder="Enter value"
    placeholderTextColor="#999"
    style={
      {
        "backgroundColor": "transparent",
        "borderBottomColor": "transparent",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "borderBottomWidth": 1,
        "borderLeftColor": "transparent",
        "borderLeftWidth": 1,
        "borderRightColor": "transparent",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "transparent",
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "borderTopWidth": 1,
        "color": "textPrimary",
        "flex": 1,
        "fontFamily": "NunitoSans_7pt-Medium",
        "fontSize": 16,
        "height": 66,
        "letterSpacing": -0.5,
        "minWidth": 0,
        "paddingBottom": 14,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 26,
      }
    }
    testID="input"
  />
</View>
`;

exports[`Input should render Input component with variant outlined 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "position": "relative",
      "zIndex": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#808080",
        "fontFamily": "NunitoSans_7pt-Medium",
        "fontSize": 12,
        "left": 16,
        "position": "absolute",
        "top": 8,
        "zIndex": 200,
      }
    }
    suppressHighlighting={true}
  >
    Name
  </Text>
  <TextInput
    allowFontScaling={false}
    focusVisibleStyle={{}}
    focusable={true}
    onChangeText={[Function]}
    placeholder="Enter value"
    placeholderTextColor="#999"
    style={
      {
        "backgroundColor": "#fff",
        "borderBottomColor": "#dbdbdb",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "borderBottomWidth": 1,
        "borderLeftColor": "#dbdbdb",
        "borderLeftWidth": 1,
        "borderRightColor": "#dbdbdb",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "#dbdbdb",
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "borderTopWidth": 1,
        "color": "textPrimary",
        "flex": 1,
        "fontFamily": "NunitoSans_7pt-Medium",
        "fontSize": 16,
        "height": 66,
        "letterSpacing": -0.5,
        "minWidth": 0,
        "paddingBottom": 14,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 26,
      }
    }
    testID="input"
  />
</View>
`;
